Source/Syrup/Player/SyrupPlayerCharacter.h

```diff
@@ -34,8 +37,28 @@ class SYRUP_API ASyrupPlayerCharacter : public ACharacter
 	
 	/* /\ Components /\ *\
 	\* ---------------- */
+	
+	/* ----------------- *\
+	\* \/  Resources  \/ */
+private:
+	//Energy of the Character
+	UPROPERTY(EditAnywhere, Category = "Resources")
+	int Energy;
+
+	/* /\  Resources  /\ *\
+	\* ----------------- */
+
+	/* ------------------------ *\
+	\* \/  Plant Management  \/ */
+public:
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Plant")
+	int PlantSelected;
 
+	UPROPERTY(EditAnywhere, Category = "Plant")
+	TArray<TSubclassOf<APlant>> PlantTypes;
```

Comment:
Should be private and have add and remove functions to ensure there are never duplicates and that null is never added. If you still want it to be setable from the details panel, you can create a second protected EditDefaultsOnly variable of type `TSet<TSubclassOf<APlant>>` which will prevent duplicates from being added, and OnConstruct() iterate over the set and add each of its elements (if not null) to PlantTypes.

Also, plant types are not the most descriptive of names, perhaps rename to Plantable Types.

Also needs comment.